pkg/stream/edgedconnection.go

```diff
@@ -8,5 +8,7 @@ type EdgedConnection interface {
 	Serve(tunnel SafeWriteTunneler) error
 	CacheTunnelMessage(msg *Message)
 	GetMessageID() uint64
+	CloseReadChannel()
+	CleanChannel()
```

Comment:
Why do we need CleanChannel function？

Reply:
To make sure the stop channel won't block the sender go routine.